api/v1/storageclusterpeer_types.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2020 Red Hat OpenShift Container Storage.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+type NamespacedName struct {
+	Name      string `json:"name"`
+	Namespace string `json:"namespace"`
+}
+
+// RemoteClusterSpec specifies the spec required for the remote cluster
+type RemoteClusterSpec struct {
+	// APIEndpoint is the URI of the ODF api server
+	ApiEndpoint string `json:"apiEndpoint"`
+
+	// OnboardingTicket holds an identity information required by the local ODF cluster to onboard.
+	OnboardingTicket string `json:"onboardingTicket"`
+
+	// ClusterName holds the namespacedName of the Remote ODF Cluster
+	ClusterName NamespacedName `json:"clusterName"`
+}
+
+// BlockPoolMirroringSpec enables setting up of mirroring for blockPools in the same namespace.
+type BlockPoolMirroringSpec struct {
+	// LabelSelector is used to select blockPools by label
+	Selector *metav1.LabelSelector `json:"selector"`
+}
+
+// StorageClusterPeerSpec defines the desired state of StorageClusterPeer
+type StorageClusterPeerSpec struct {
+
+	// RemoteCluster specifies the spec required for the remote cluster
+	RemoteCluster RemoteClusterSpec `json:"remoteCluster"`
+
+	// LocalClusterName holds the Name of the local ODF Cluster
+	LocalClusterName corev1.LocalObjectReference `json:"localClusterName"`
```

Comment:
Let's try to align the spec structure between RemoteCluster and LocalCluster
 ```suggestion
	LocalCluster LocalClusterSpec `json:"localCluster"`
```

```
type LocalClusterSpec struct {
    Name: corev1.LocalObjectReference `json:"name"`
}
```